d2/src/test/java/com/linkedin/d2/balancer/strategies/relative/PartitionStateDataBuilder.java

```diff
@@ -0,0 +1,93 @@
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.d2.balancer.strategies.relative;
+
+import com.linkedin.d2.balancer.clients.TrackerClient;
+import com.linkedin.d2.balancer.strategies.degrader.LoadBalancerQuarantine;
+import com.linkedin.d2.balancer.strategies.degrader.RingFactory;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * The helper class that builds an object of {@link PartitionState}
+ */
+public class PartitionStateDataBuilder {
```

Comment:
This is only for test, it is under test module already.

Reply:
I didn't really name it as PartitionStateBuilder because it can be confusing if one day we have an actual builder for PartitionState, the package name will be same for test data builder and non-test builder.
I can rename it to PartitionStateTestDateBuilder if that is more clear.